### ISSUE TITLE
LaunchApplicationAsync() throws 'Unauthorized' exception

### ISSUE THREAD

**Issue by DavidRatliff** [Microsoft/WindowsDevicePortalWrapper#279]
I am attempting to remotely launch a Hololens application. My code, as posted below, throws an 'Unauthorized' exception. To my understanding, this exception is used for missing or incorrect credentials, but that cannot be the case, as the portal connection opens successfully. I am running this project as a UWP from the Unity Editor. Any help is greatly appreciated!

```
private async Task LaunchApplication()
    {
        try
        {
            Debug.Log("Launching...");
            await portal.LaunchApplicationAsync(appid, packageName);
            Debug.Log("Launched");
        }
        catch (DevicePortalException e)
        {
            Debug.Log(e.Reason);
        }
    }
```

**Comment by hpsin** [comment]
Hi David,

Is there any chance you're running this with the connection address at loopback/localhost?

Does every app you're attempting to launch fail with this?

Thanks,
Hirsch
________________________________

**Comment by DavidRatliff** [comment]
I have tried connecting at localhost and via internet, and with several different apps. All produce this exception. Thanks for the response!

**Comment by hpsin** [comment]
Hm - do the other REST APIs work? e.g. you can get the device family? One thing to be aware of is that there might be two LaunchApplication APIs, one of which is for IoT.

**Comment by DavidRatliff** [comment]
Yes, upon connection I can successfully retrieve the Hololens IP, OS version, and device family.

If I'm reading the [documentation](https://docs.microsoft.com/en-us/windows/uwp/debug-test-perf/device-portal-api-core#task-manager) correctly, there is only a single LaunchApplication request type, used for both Hololens and IoT.

**Comment by hpsin** [comment]
Ah, yes, you're correct.  Can you share how you're getting the AppId and the packageName?

**Comment by DavidRatliff** [comment]
Apologies for the inactivity, it's been finals season for me. I'm pulling that information manually from the response to the /api/app/packagemanager/packages request. For example:

appid = "Microsoft.MicrosoftEdge_8wekyb3d8bbwe!MicrosoftEdge"
packageName = "Microsoft.MicrosoftEdge_38.14393.1770.0_neutral__8wekyb3d8bbwe"

**Comment by smstuebe** [comment]
I got it working by changing the auth to basic auth.

in HttpHeadersHelpers I added

    private void ApplyHttpHeaders(
        HttpClient client,
        HttpMethods method)
    {
        var byteArray = Encoding.ASCII.GetBytes($"{deviceConnection.Credentials.UserName}:{deviceConnection.Credentials.Password}");
        client.DefaultRequestHeaders.Authorization = new AuthenticationHeaderValue("Basic", Convert.ToBase64String(byteArray));
        this.ApplyUserAgentHeader(client);
        this.ApplyCSRFHeader(client, method);
    }

and I removed the credential stuff from the HttpClientHandler initializations

    new HttpClientHandler
    {
         //UseDefaultCredentials = false,
         //Credentials = this.deviceConnection.Credentials,
        ....
    }
